### PR TITLE
Implements back end game logic

### PR DESCRIPTION
**Pull request by PThompson284**
Implements the rules of the game and allows you to play
#19 #20

DEVELOPMENT BRANCH PR

### CODE REVIEW

**Review by DavidWellsTheDeveloper — APPROVED** [review]

File: server/src/Game.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    public Move move = new Move(this);
+    public String winner = "";
+
+    public Game(String playerOneName, String playerTwoName){
+
+        playerOne = playerOneName;
+        playerTwo = playerTwoName;
+        turn = "blue";
+
+        for(int i=0; i<9; i++){
```
Comment (tmsdbqk) [inline server/src/Game.java:35]: Suggest changing magic numbers and indices to something more descriptive.

File: server/src/Game.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        if(isTurn(player)){
+            if(move.selectedCol == col && move.selectedRow == row){ // if the second half of the move is the same as the first half (re-selecting the same tile)
+                move.selectedCol = -1;
+                move.selectedRow = -1;
+            }
+            else if(move.isFriendlyUnit(row, col)){ // if the player is selecting one of their own tiles
+                move.selectedCol = col;
+                move.selectedRow = row;
+            }
+            else if(move.validTiles[row][col].equals("*")) { // if row, col is a valid tile to move to
```
Comment (tmsdbqk) [inline server/src/Game.java:77]: What if for some reason the row or column has an index out of bounds of the board?

File: server/src/Game.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        System.out.println("");
+        printBoard();
+    }
+
+    private boolean isTurn(String player){
+        // returns true if it is the turn of the player submitting the move, else returns false.
+        return (turn.equals("blue") && player.equals(playerOne)) || (turn.equals("red") && player.equals(playerTwo));
+    }
+
+    private void checkIfWinner(){
```
Comment (tmsdbqk) [inline server/src/Game.java:111]: We might want some of these methods to return an integer or flag of some sort, so the server knows when to make an update to the database without having to call the object again. Return -1 if no winner, or the player ID if winner, etc. We can talk about it more.

File: server/src/Move.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    void updateValidTiles(){
+        for(int i=0; i<9; i++) {
+            for (int j = 0; j < 7; j++) {
+                validTiles[i][j] = "_";
+            }
+        }
+        if(selectedRow != -1 && selectedCol != -1) {
+            if(selectedCol <= 5) handleUpdate(selectedRow, selectedCol+1);
```
Comment (tmsdbqk) [inline server/src/Move.java:33]: What do these numbers mean?

**Pull request merged**